server/src/test/java/io/crate/planner/operators/JoinPlanBuilderTest.java

```diff
@@ -179,4 +178,33 @@ public void testImplicitToExplicit_OrderOfPairsRemains() {
             assertThat(newPairAtPos.right()).isEqualTo(oldPairAtPos.right());
         }
     }
+
+    @Test
+    public void test_implicit_join_pairs_from_join_conditions() {
+        List<JoinPair> joinPairs = new ArrayList<>();
+        joinPairs.add(JoinPair.of(T3.T1, T3.T2, JoinType.CROSS, null));
+        joinPairs.add(JoinPair.of(T3.T2, T3.T3, JoinType.CROSS, null));
+        joinPairs.add(JoinPair.of(T3.T3, T3.T4, JoinType.INNER, asSymbol("t4.id = t3.z AND t4.id = t2.y")));
+        joinPairs.add(JoinPair.of(T3.T4, T3.T5, JoinType.RIGHT, asSymbol("t5.i = t4.id AND t5.i = t1.x")));
```

Comment:
Do we need 5 tables to test the issue? The changes entry implies to me that joining 3 tables are enough?

Reply:
Since we're testing here the method directly, I added a more complex test which also shows that the join type is kept from the original `JoinPair` when part of the condition is split and becomes the join condition of a new `JoinPair`.

Reply:
I've added a simpler test to show a complete join query plan tree.